### ISSUE TITLE
Call function feature request

### ISSUE THREAD

**Issue by minoism** [hendraanggrian/kotlinpoet-ktx#2]
For {}, have `beginFlow`. But for (), may should add an easier way to generate code?

```kotlin
fun save(tag: String, vararg data: Any) {}
```
Generated:
```kotlin
save("foo", a, b, c, d)
```
I may need to do this sometimes:
```kotlin
append("save(")
append("%S, ", "foo")
append("a, ")
append("b, ")
append("c, ")
append("d")
append(")")
```

So is there a way to simplify these steps?
```kotlin
appendCall("save") {
  appendLine("%S, ", "foo")
  appendLine("a")
  appendLine("a")
  appendLine("b")
  appendLine("c")
  appendLine("d")
}
// or simply...
appendCall("save") {
  +"%S".format("foo")
  +"a"
  +"b"
  +"c"
  +"d"
}
```

**Comment by hanggrian** [comment]
I don't quite understand this one. What's wrong with `appendLine("save(%S, a, b, c, d)", "foo")`?

You can even do:

```kotlin
appendLine {
    append("save(%S, a, b, c, d)", "foo")
    ...
}
```

**Comment by minoism** [comment]
> I don't quite understand this one. What's wrong with `appendLine("save(%S, a, b, c, d)", "foo")`?
> 
> You can even do:
> 
> ```kotlin
> appendLine {
>     append("save(%S, a, b, c, d)", "foo")
>     ...
> }
> ```

This is not wrong, because it is hard-coded, but if the parameters I need to pass in may be from other arrays, or in some special cases I need to enter later, then this may be a bit tricky

Like: 
```Kotlin
val parameters = ...
append {
  append("save(")
  append("%S, ", "foo")
  parameters.forEachIndexed { index, any ->
    if (index != parameters.lastIndex) {
      append("$any, ")
    } else {
      append("$any")
    }
  }
  append(")")
}
```

**Comment by hanggrian** [comment]
> ```kotlin
> val parameters = ...
> append {
>   append("save(")
>   append("%S, ", "foo")
>   parameters.forEachIndexed { index, any ->
>     if (index != parameters.lastIndex) {
>       append("$any, ")
>     } else {
>       append("$any")
>     }
>   }
>   append(")")
> }
> ```

This is similar to how I'm currently using this library. It's not pretty, but the API is made to follow `StringBuilder`. The question is, how would `appendCall` be of any help here?

**Comment by minoism** [comment]
> The question is, how would `appendCall` be of any help here?

I think it can improve the conciseness of the code, such as commas and parentheses, can be simplified by this method, without manual process.

**Comment by dalewking** [comment]
Here is my workaround where `parameterNames` is a list of parameter names (in your case `val parameterNames = listOf( "a", "b", "c", "d")`

```
      .addStatement(
          parameterNames.joinToString(", ","save(", ")") { "%L" },
          *parameterNames.toTypedArray()
      )
```

will generate

```
    save(a, b, c, d)
```
